src/jesse_json_path.erl

```diff
@@ -9,13 +9,17 @@
 %% API for path management
 -export([new/0, new/1, push/2, pop/1, to_string/1]).
 
+-ifdef(deprecated_types).
+-type dict() :: dict:dict().
+-endif.
+
 -type elem_key_type() :: atom | binary | string | undefined.
 -type elem_type() :: list | elem_key_type().
 -type kvc_obj() :: kvc_obj_node() | [kvc_obj_node()] | list().
 -type kvc_key() :: binary() | atom() | string().
 -type proplist() :: [{kvc_key(), kvc_obj()}].
 -type kvc_obj_node() :: proplist() | {struct, proplist()}
-                      | dict:dict() | gb_tree:tree() | term().
+                      | dict() | gb_tree:tree() | term().
```

Comment:
@zdaniel86  I don't understand why it compiles, but please change to  `gb_trees:tree()`: in both https://github.com/erlang/otp/blob/OTP-19.1/lib/stdlib/src/gb_trees.erl#L160  there is such module as `gb_trees`, but there is definitely no such things as `gb_tree`